src/config/sled.rs

```diff
@@ -460,6 +464,112 @@ impl IdentityKeyStore for SledConfigStore {
     }
 }
 
+fn thread_key(t: &Thread) -> Vec<u8> {
+    let mut bytes = SLED_TREE_THREAD_PREFIX.as_bytes().to_owned();
+    bytes.append(&mut t.into());
+    bytes
+}
+
+impl MessageStore for SledConfigStore {
+    type MessagesIter = SledMessagesIter;
+    fn save_message(
+        &mut self,
+        message: libsignal_service::prelude::Content,
+        receiver: Option<impl Into<ServiceAddress>>,
+    ) -> Result<(), Error> {
+        let receiver = receiver.map(|s| s.into());
+
+        let id = MessageIdentity::try_from(&message)?;
+        let thread = Thread::from_content_receiver(&message, receiver.as_ref())?;
+        let timestamp = id.1;
+
+        log::trace!("Storing a message with id: {:?}", id);
+
+        let tree_messages = self
```

Comment:
Do you have a real need for storing and accessing messages one by one? IMO, we should only store messages once.